### PR TITLE
LPS-194762 A Master Page in draft can be mark as default

### PR DESCRIPTION
**Pull request by BarbaraCabrera**
## Motivation

[Link to ticket](https://liferay.atlassian.net/browse/LPS-194762)

## Proposed solution

Modify the CopyHelper to set the "published" property to false when copying Master Layouts and Utility Pages.

## Test Scenario 1

1. Navigate to Design > Master Pages(MP)
2. Create a new MP and publish it
3. Make a copy from that MP
4. Check that you can’t mark the copy by default because it’s in draft and it has never been publish before.

image

## Test Scenario 2

1. Navigate to Pages > Utility Pages(UP)
2. Make a copy from any UP existing
3. Check that you can’t mark the copy by default because it’s in draft and it has never been publish before.

Pasted Graphic 3

## Notes

If we edit the Master Page/Utility Page and leave it in draft, but it has a previous version published, it will let us mark it by default.

### CODE REVIEW

**Comment by BarbaraCabrera** [comment]
ci:test:sf

File: modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/portlet/action/CopyLayoutPageTemplateEntryMVCActionCommand.java [hunk 1]
```diff
@@ -91,29 +91,24 @@ private LayoutPageTemplateEntry _copyLayoutPageTemplateEntry(
...
 
-		Layout draftSourceLayout = _layoutLocalService.fetchLayout(
-			_portal.getClassNameId(Layout.class), sourceLayout.getPlid());
-
 		Layout targetLayout = _layoutLocalService.getLayout(
 			layoutPageTemplateEntry.getPlid());
 
-		Layout draftTargetLayout = _layoutLocalService.fetchLayout(
-			_portal.getClassNameId(Layout.class), targetLayout.getPlid());
+		Layout draftTargetLayout = targetLayout.fetchDraftLayout();
```
Comment (ealonso) [inline modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/portlet/action/CopyLayoutPageTemplateEntryMVCActionCommand.java:109]: @BarbaraCabrera you can inline it

File: modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/servlet/taglib/util/MasterLayoutActionDropdownItemsProvider.java [hunk 2]
```diff
@@ -433,6 +434,7 @@ private String _getItemSelectorURL() {
 					"layoutPageTemplateEntryId",
 					_layoutPageTemplateEntry.getLayoutPageTemplateEntryId()
 				).buildString());
+			dropdownItem.setDisabled(!_layout.isPublished());
```
Comment (ealonso) [inline modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/servlet/taglib/util/MasterLayoutActionDropdownItemsProvider.java:437]: @BarbaraCabrera it should be ordered, like in the other cases

**Comment by BarbaraCabrera** [comment]
> @BarbaraCabrera I think the proposed solution is from other tasks, right?

Yes, I have followed the same behaviour as the other cases.

File: modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/servlet/taglib/util/MasterLayoutActionDropdownItemsProvider.java (hunk 2 above)
Comment (BarbaraCabrera) [reply modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/servlet/taglib/util/MasterLayoutActionDropdownItemsProvider.java:437]: When I pass the SF it puts it back on this line, so I can't apply this change

**Comment by ealonso** [comment]
ci:forward

**Comment by Tim-Cao** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/140195

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**